tests/integration/test_metrics.py

```diff
@@ -650,6 +675,87 @@ def test_transaction_metrics(
         "type": "d",
         "value": [9.910106, 9.910106],
     }
+    assert metrics["c:transactions/count_per_root_project@none"] == {
+        "timestamp": int(timestamp.timestamp()),
+        "org_id": 1,
+        "project_id": 42,
+        "retention_days": 90,
+        "tags": {"transaction": "transaction_which_starts_trace"},
+        "name": "c:transactions/count_per_root_project@none",
+        "type": "c",
+        "value": 2.0,
+    }
+
+
+def test_transaction_metrics_count_per_root_project(
+    mini_sentry,
+    relay,
+    relay_with_processing,
+    metrics_consumer,
+    transactions_consumer,
+):
+    metrics_consumer = metrics_consumer()
+    transactions_consumer = transactions_consumer()
+
+    relay = relay_with_processing(options=TEST_CONFIG)
+
+    project_id = 42
+    mini_sentry.add_full_project_config(41)
+    mini_sentry.add_full_project_config(project_id)
+    timestamp = datetime.now(tz=timezone.utc)
+
+    for project_id in (41, 42):
+        config = mini_sentry.project_configs[project_id]["config"]
+        config["sessionMetrics"] = {"version": 1}
+        config["breakdownsV2"] = {
+            "span_ops": {"type": "spanOperations", "matches": ["react.mount"]}
+        }
+        config["transactionMetrics"] = {
+            "version": 1,
+        }
+
+    transaction = generate_transaction_item()
+    transaction["timestamp"] = timestamp.isoformat()
+    transaction["measurements"] = {
+        "foo": {"value": 1.2},
+        "bar": {"value": 1.3},
+    }
+    relay.send_transaction(41, transaction, transaction_from_dsc="test")
```

Comment:
Instead of sending this transaction to project 41, I would send it to project 42 with `project_id=41` in the DSC. And then verify that the metric has project 41, not 42. This is what I meant originally with

> a test with a DSC where the DSC's project ID is different from the event's project ID.